From e81963b2601ab0b01559964e4e615a83076c9052 Mon Sep 17 00:00:00 2001
From: Soren Brinkmann <soren.brinkmann@xilinx.com>
Date: Mon, 10 Dec 2012 13:55:47 -0800
Subject: [PATCH 498/628] usb: xilinxps_udc: Use identifier for memory
 allocation

git://github.com/Xilinx/linux-xlnx.git xilinx-v14.7
commit 4c4a3c256db62371f6efd1ef68c1432984e7ba9f

The argument for sizeof in memory allocators should use
the *<identifier> form, rather than <type>.

Signed-off-by: Soren Brinkmann <soren.brinkmann@xilinx.com>
Acked-by: Michal Simek <michal.simek@xilinx.com>
Signed-off-by: Liming Wang <liming.wang@windriver.com>
---
 drivers/usb/gadget/xilinx_usbps_udc.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/usb/gadget/xilinx_usbps_udc.c b/drivers/usb/gadget/xilinx_usbps_udc.c
index 6a939da..9c9ae70 100644
--- a/drivers/usb/gadget/xilinx_usbps_udc.c
+++ b/drivers/usb/gadget/xilinx_usbps_udc.c
@@ -2627,7 +2627,7 @@ static int __devinit struct_udc_setup(struct xusbps_udc *udc,
 	pdata = pdev->dev.platform_data;
 	udc->phy_mode = pdata->phy_mode;
 
-	udc->eps = kzalloc(sizeof(struct xusbps_ep) * udc->max_ep, GFP_KERNEL);
+	udc->eps = kzalloc(sizeof(*udc->eps) * udc->max_ep, GFP_KERNEL);
 	if (!udc->eps) {
 		dev_err(&pdev->dev, "malloc xusbps_ep failed\n");
 		return -1;
@@ -2724,7 +2724,7 @@ static int __devinit xusbps_udc_probe(struct platform_device *pdev)
 		return -ENODEV;
 	}
 
-	udc_controller = kzalloc(sizeof(struct xusbps_udc), GFP_KERNEL);
+	udc_controller = kzalloc(sizeof(*udc_controller), GFP_KERNEL);
 	if (udc_controller == NULL) {
 		dev_err(&pdev->dev, "malloc udc failed\n");
 		return -ENOMEM;
-- 
1.7.5.4

